model/change_point_detection.go

```diff
@@ -23,23 +23,25 @@ var (
 )
 
 type ChangePoint struct {
-	Index        int           `bson:"index" json:"index" yaml:"index"`
-	Measurement  string        `bson:"measurement" json:"measurement" yaml:"measurement"`
-	CalculatedOn time.Time     `bson:"calculated_on" json:"calculated_on" yaml:"calculated_on"`
-	Algorithm    AlgorithmInfo `bson:"algorithm" json:"algorithm" yaml:"algorithm"`
-	Triage       TriageInfo    `bson:"triage" json:"triage" yaml:"triage"`
+	Index         int           `bson:"index" json:"index" yaml:"index"`
+	Measurement   string        `bson:"measurement" json:"measurement" yaml:"measurement"`
+	CalculatedOn  time.Time     `bson:"calculated_on" json:"calculated_on" yaml:"calculated_on"`
+	Algorithm     AlgorithmInfo `bson:"algorithm" json:"algorithm" yaml:"algorithm"`
+	Triage        TriageInfo    `bson:"triage" json:"triage" yaml:"triage"`
+	PercentChange float64       `bson:"percent_change" json:"percent_change" yaml:"percent_change"`
 }
 
-func CreateChangePoint(index int, measurement string, algorithmName string, algorithmVersion int, algoOptions []AlgorithmOption) ChangePoint {
+func CreateChangePoint(index int, measurement string, algorithmName string, algorithmVersion int, algoOptions []AlgorithmOption, percentChange float64) ChangePoint {
```

Comment:
Apologize for being late in the game here, but what is the point of having this `CreateChangePoint` function? Aside from setting `CalculatedOn` to `time.Now()` there is no other setup, nor are there private fields. I think it would be cleaner to create instances of struct directly.

Reply:
This function already existed and I was just adding to it. I don't have any hard opinion on it. I can remove it and instantiate the struct directly, unless there was a specific reason for this existing, @tychoish @ancostas ?

Reply:
This is a constructor to ensure `CalculatedOn` and `Triage` are set up in a consistent way.

Reply:
Seems reasonable, I think its mostly how many parameters passed in that bother me. It's too late anyway to make any significant changes to the model. I will mark as resolved.